python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py

```diff
@@ -0,0 +1,294 @@
+
+import asyncio
+import time as time
+
+import yaml
+
+from lsst.ts.idl.enums import ATWhiteLight
+
+from lsst.ts import salobj
+
+
+class PowerOnATCalSys(salobj.BaseScript):
+    """Powers on the ATCalSys dome flat illuminator
+    (ATWhiteLight and ATMonochromator)
+    required to perform image calibrations over white light.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    """
+
+    def __init__(self, index, add_remotes: bool = True):
+        super().__init__(
+            index=index,
+            descr="Power On AT Calibration System ",
+        )
+
+        self.white_light_source = None
+        self.monochromator = None
+
+        # White lamp config
+        self.timeout_lamp_warm_up = 60 * 15
+        self.track_lamp_warmup = 60
+        self.cmd_timeout = 30
+
+        # Chiller config
+        self.timeout_chiller_cool_down = 60 * 10
+        self.chiller_temp_tolerance_relative = 0.1
+
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_standardscripts/auxtel/calibrations/power_on_atcalsys.yaml
+            title: PowerOnATCalSys v1
+            description: Configuration for PowerOnATCalSys.
+              Each attribute can be specified as a scalar or array.
+              All arrays must have the same length (one item per image).
+            type: object
+            properties:
+              chiller_temperature:
+                description: Set temperature for the chiller
+                type: number
+                default: 20
+                minimum: 15
+
+              whitelight_power:
+                description: White light power.
+                type: number
+                default: 910
+                minimum: 0
+
+              wavelength:
+                description: Wavelength (nm). 0 nm is for white light.
+                type: number
+                default: 0
+                minimum: 0
+
+              grating_type:
+                description: Grating type for each image. The choices are 0=blue, 1=red, 2=mirror.
+                type: integer
+                enum: [0, 1, 2]
+                default: 2
+
+              entrance_slit_width:
+                description: Width of the monochrometer entrance slit (mm)
+                type: number
+                minimum: 0
+                default: 7
+
+              exit_slit_width:
+                description: Width of the monochromator entrance slit (mm)
+                type: number
+                minimum: 0
+                default: 7
+
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config : ``self.cmd_configure.DataType``
+
+        Raises
+        ------
+        salobj.ExpectedError :
```

Comment:
Usually we only document the exceptions that the method itself raises. This one in particular is raised by another method that does the configuration validation before calling `configure`. You should probably remove this here.